distributed/shuffle/_worker_extension.py

```diff
@@ -46,108 +57,54 @@ class ShuffleClosedError(RuntimeError):
     pass
 
 
-class ShuffleRun:
-    """State for a single active shuffle execution
-
-    This object is responsible for splitting, sending, receiving and combining
-    data shards.
-
-    It is entirely agnostic to the distributed system and can perform a shuffle
-    with other `Shuffle` instances using `rpc` and `broadcast`.
-
-    The user of this needs to guarantee that only `Shuffle`s of the same unique
-    `ShuffleID` interact.
-
-    Parameters
-    ----------
-    worker_for:
-        A mapping partition_id -> worker_address.
-    output_workers:
-        A set of all participating worker (addresses).
-    column:
-        The data column we split the input partition by.
-    schema:
-        The schema of the payload data.
-    id:
-        A unique `ShuffleID` this belongs to.
-    run_id:
-        A unique identifier of the specific execution of the shuffle this belongs to.
-    local_address:
-        The local address this Shuffle can be contacted by using `rpc`.
-    directory:
-        The scratch directory to buffer data in.
-    nthreads:
-        How many background threads to use for compute.
-    loop:
-        The event loop.
-    rpc:
-        A callable returning a PooledRPCCall to contact other Shuffle instances.
-        Typically a ConnectionPool.
-    barrier:
-        A PooledRPCCall to to contact the scheduler.
-    memory_limiter_disk:
-    memory_limiter_comm:
-        A ``ResourceLimiter`` limiting the total amount of memory used in either
-        buffer.
-    """
-
+class ShuffleRun(Generic[TransferShardIDType, PartitionIDType, PartitionType]):
```

Comment:
Nit: In a follow up PR we should move these classes out of this module. They are only indirectly connected to the actual worker extension. For instance, I could even see these classes being used to implement a non-distributed version of rechunking/shuffling in dask/dask eventually to replace partd.

Reply:
Agreed, I'll add a ticket for this.